### ISSUE TITLE
Global Endpoints Controller options 

### ISSUE THREAD

**Issue by bobholt** [endpoints/endpoints#51]
We're entering the territory of allowing options in endpoints that users may want to configure at an API level.

For example, allowing client-generated IDs when creating models. #50 allows this, but some APIs may reasonably not want to.

This is an option that shouldn't be configured on a resource-by-resource basis, but globally for the entire API. I'm imagining a pattern like this:

```
// classes/my-controller.js
const Endpoints = require('endpoints');
export Endpoints.Controller.extend({
  clientGeneratedIds: false
});
```

```
// modules/some-controller.js
const MyController = require('../classes/my-controller');
module.exports = new MyController({
  source: new Endpoints.BookshelfSource({
    model: require('./model')
  })
});
```

Thoughts? Better options?

**Comment by tkellen** [comment]
I'd rather the controllers not be aware of their containing application... maybe we could allow you to extend a controller with properties and then re-use it elsewhere?

**Comment by tkellen** [comment]
This description is exactly what my inclination to do would be. It's been a long time since I looked at libraries that make it easy to do extend multiple times in JS. Do you know of any we can leverage offhand? If no, maybe we should ask the coop?

**Comment by bobholt** [comment]
All I've really done with this pattern is with Backbone: https://github.com/jashkenas/backbone/blob/master/backbone.js#L1778-L1812

**Comment by tkellen** [comment]
resolved by #102
